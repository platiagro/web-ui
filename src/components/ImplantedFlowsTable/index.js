import React from 'react';

import { Table, Typography, Tooltip } from 'antd';
import { Link } from 'react-router-dom';

import './style.scss';

const { Paragraph } = Typography;

const tableColumns = [
  {
    title: 'Nome',
    dataIndex: 'flowName',
    key: 'flowName',
  },
  {
    title: 'URL',
    dataIndex: 'url',
    key: 'url',
    render: (value) => (
      <Tooltip title={value}>
        <Paragraph ellipsis copyable>
          {value}
        </Paragraph>
      </Tooltip>
    ),
  },
  {
    title: 'Data de Criação',
    dataIndex: 'created',
    key: 'created',
    // render: (value) => new Date(value).toLocaleString(),
  },
  {
    title: 'Ação',
    dataIndex: 'action',
    key: 'action',
    render: (value) => (
      <a target='_blank' href='/notebook/kubeflow-anonymous/server-1/tree?'>
        Monitoramento
      </a>
    ),
  },
];

const ImplantedFlowsTable = ({ flowList }) => {
  return (
    <Table
      // rowKey={(record) => record.uuid}
      dataSource={flowList}
      columns={tableColumns}
      pagination={{ pageSize: 9 }}
    // scroll={{ y: 'calc(100vh - 480px)' }}
    />
  );
};

export default ImplantedFlowsTable;
