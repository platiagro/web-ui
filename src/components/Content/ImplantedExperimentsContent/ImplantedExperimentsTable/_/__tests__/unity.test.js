import React from 'react';
import { Table } from 'antd';
import { shallow, mount } from 'enzyme';
import ImplantedFlowsTable from '..';

const implatedFlows = [
  {
    key: 'key 1',
    flowName: 'flowName',
    url: 'url',
    created: 'created',
    status: 'Succeeded',
    action: 'action',
  },
  {
    key: 'key 2',
    flowName: 'flowName',
    url: 'url',
    created: 'created',
    status: 'Failed',
    action: 'action',
  },
];

describe('ImplantedFlowsTable component', () => {
  it('is expected render without crashing', () => {
    shallow(<ImplantedFlowsTable flowList={implatedFlows} />);
  });

  it('is expected render html correctly', () => {
    const wrapper = shallow(<ImplantedFlowsTable flowList={implatedFlows} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('is expected to exist Table', () => {
    const wrapper = mount(<ImplantedFlowsTable flowList={implatedFlows} />);
    expect(wrapper.find(Table).exists()).toBeTruthy();
  });
});
