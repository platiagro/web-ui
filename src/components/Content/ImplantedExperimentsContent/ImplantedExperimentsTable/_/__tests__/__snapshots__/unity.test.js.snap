// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImplantedFlowsTable component is expected render html correctly 1`] = `
<withStore(Table)
  columns={
    Array [
      Object {
        "dataIndex": "flowName",
        "key": "flowName",
        "title": "Nome",
      },
      Object {
        "dataIndex": "url",
        "key": "url",
        "render": [Function],
        "title": "URL",
      },
      Object {
        "dataIndex": "created",
        "key": "created",
        "title": "Data de Criação",
      },
      Object {
        "dataIndex": "deployStatus",
        "key": "deployStatus",
        "render": [Function],
        "title": "Status",
      },
      Object {
        "dataIndex": "action",
        "key": "action",
        "render": [Function],
        "title": "Ação",
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "action": "action",
        "created": "created",
        "flowName": "flowName",
        "key": "key 1",
        "status": "Succeeded",
        "url": "url",
      },
      Object {
        "action": "action",
        "created": "created",
        "flowName": "flowName",
        "key": "key 2",
        "status": "Failed",
        "url": "url",
      },
    ]
  }
  pagination={
    Object {
      "pageSize": 9,
    }
  }
/>
`;
